### PR TITLE
Update ``earthaccess`` example to use Coiled Functions

### PR DESCRIPTION
**Pull request by jrbourbeau**
This is more inline with the usage pattern we're seeing in the wild. The example currently on `main` is still valid, but we have other `xarray` example already here (can always add it back in the future).

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -20,5 +19,4 @@ dependencies:
   - zarr
   - h5netcdf
   - pre-commit
-  - pip:
-    - git+https://github.com/nsidc/earthaccess.git@main
\ No newline at end of file
+  - earthaccess
```
Comment (jrbourbeau) [inline environment.yml:22]: We can use the latest `earthaccess` release now

**Comment by jrbourbeau** [comment]
Docker builds are failing. I suspect they're unrelated to the changes here. Running those builds against `main` here https://github.com/coiled/examples/actions/runs/6408111620 to know for sure

**Comment by ntabris** [comment]
> Docker builds are failing

Maybe `libmamba filesystem error: cannot create directories: No space left on device [/opt/conda/pkgs/libnvjpeg-11.9.0.86-0]` is the issue?

**Review by mrocklin — APPROVED** [review]
Cool.  Looks good to me.  As someone who might have to demo this and doesn't know much about it I would welcome a few more details somewhere about the application and files (what are we computing, how much data is it?) but that can certainly be follow-up work (with some forever-put-off date).  It's not that important or urgent.  

Good to go from my perspective.

**Pull request merged**